pkg/spdk/engine.go

```diff
@@ -1234,6 +1234,19 @@ func (e *Engine) ReplicaDelete(spdkClient *spdkclient.Client, replicaName, repli
 		return fmt.Errorf("replica %s recorded address %s does not match the input address %s for engine %s replica delete", replicaName, e.ReplicaAddressMap[replicaName], replicaAddress, e.Name)
 	}
 
+	// Pause the IO and flush cache by suspending the NVMe initiator
+	if e.Frontend == types.FrontendSPDKTCPBlockdev && e.Endpoint != "" {
+		// The system-created snapshot during a rebuilding does not need to guarantee the integrity of the filesystem.
+		if err = e.initiator.Suspend(true, true); err != nil {
+			return errors.Wrapf(err, "failed to suspend NVMe initiator before removing replica %s base bdev from engine %s RAID", replicaName, e.Name)
+		}
+		defer func() {
+			if err = e.initiator.Resume(); err != nil {
+				err = errors.Wrapf(err, "failed to suspend NVMe initiator after removing replica %s base bdev from engine %s RAID", replicaName, e.Name)
+			}
+		}()
+	}
+
```

Comment:
What do we need to suspend frontend for the replica deletion?

Reply:
Let's remove this commit. 
When Longhorn deletes a replica from the engine, spdk_tgt will print out lots of errors like `nvme_tcp.c:2177:nvme_tcp_qpair_process_completions: *ERROR*: Failed to flush tqpair=0x1826e70 (9): Bad file descriptor`. I was thinking adding a frontend suspend would avoid it. but after testing it today, the answer is NO.